src/ansi.rs

```diff
@@ -0,0 +1,2062 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// This module was originally part of the `alacritty_terminal` crate, which is
+// licensed under the Apache License, Version 2.0 and is part of the Alacritty
+// project (https://github.com/alacritty/alacritty).
+
+//! ANSI Terminal Stream Parsing.
+
+extern crate alloc;
+
+use alloc::borrow::ToOwned;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+
+use core::convert::TryFrom;
+use core::fmt::{self, Display, Formatter, Write};
+use core::ops::{Add, Sub};
+use core::str::FromStr;
+use core::time::Duration;
+use core::{iter, str};
+
+use log::{debug, trace};
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::{Params, ParamsIter};
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct Hyperlink {
+    /// Identifier for the given hyperlink.
+    pub id: Option<String>,
+    /// Resource identifier of the hyperlink.
+    pub uri: String,
+}
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Rgb {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+}
+
+impl Rgb {
+    /// Implementation of [W3C's luminance algorithm].
+    ///
+    /// [W3C's luminance algorithm]: https://www.w3.org/TR/WCAG20/#relativeluminancedef
+    #[cfg(feature = "floats")]
+    pub fn luminance(self) -> f64 {
+        let channel_luminance = |channel| {
+            let channel = channel as f64 / 255.;
+            if channel <= 0.03928 {
+                channel / 12.92
+            } else {
+                f64::powf((channel + 0.055) / 1.055, 2.4)
+            }
+        };
+
+        let r_luminance = channel_luminance(self.r);
+        let g_luminance = channel_luminance(self.g);
+        let b_luminance = channel_luminance(self.b);
+
+        0.2126 * r_luminance + 0.7152 * g_luminance + 0.0722 * b_luminance
+    }
+
+    /// Implementation of [W3C's contrast algorithm].
+    ///
+    /// [W3C's contrast algorithm]: https://www.w3.org/TR/WCAG20/#contrast-ratiodef
+    #[cfg(feature = "floats")]
+    pub fn contrast(self, other: Rgb) -> f64 {
+        let self_luminance = self.luminance();
+        let other_luminance = other.luminance();
+
+        let (darker, lighter) = if self_luminance > other_luminance {
+            (other_luminance, self_luminance)
+        } else {
+            (self_luminance, other_luminance)
+        };
+
+        (lighter + 0.05) / (darker + 0.05)
+    }
+}
+
+// A multiply function for Rgb, as the default dim is just *2/3.
+#[cfg(feature = "floats")]
+impl core::ops::Mul<f32> for Rgb {
```

Comment:
Do not use full paths, but instead add imports to the top of the file.

Reply:
Fixed in https://github.com/alacritty/vte/pull/85/commits/d9f17f1fe607a08bfea52ccbec26106093a08b0d.